join.spec.md

```diff
@@ -131,48 +131,42 @@ Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
 The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
 
-# Scalars
-
-Documents MUST include definitions of any scalars they use. Producers SHOULD include the `@specifiedBy` directive, but it is not mandatory.
-
-##! join__Never
+# Directives
 
-:::[definition](photos.graphql#join__Never)
+##! @join__graph
 
-The {join__Never} scalar is used to indicate the type of a field which **cannot be resolved**. This is typically due to a composition error. Producers SHOULD provide more specific error information with an {@error} directive on the relevant field.
+Declare subgraph metadata on {join__Graph} enum values. 
 
-:::[example](photos.graphql#User.favorite) -- Using {join__Never} and {@error} to mark a field error
+```graphql definition
+directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+```
 
-# Directives
+:::[example](photos.graphql#join__Graph) -- Using {@join__graph} to declare subgraph metadata on the {join__Graph} enum values.
 
 ##! @join__type
 
 Join a type to a subgraph, optionally providing an entity key.
 
 ```graphql definition
 directive @join__type(
-  graph: join__Graph!,
-  key: ref__Fragment)
-  repeatable on OBJECT
+  graph: join__Graph!
+  key: String
+) repeatable on OBJECT | INTERFACE
 ```
 
+// TODO: rename portal query?
 Keys will be passed as `representations` within a [portal query](#portal-query) to [port](#portability) a selection set between subgraphs.
 
 :::[example](photos.graphql#Image) -- Using {@join__type} to specify subgraph keys
 
-```html diagram
-<script>line => elStr(line).includes("key:")</script>
-```
-
 Multiple {@join}s can be specified for different subgraphs. It is an error to {@join} an object against the same subgraph multiple times.
 
 ##! @join__owner
 
 Specify the graph which owns the object type.
 
 ```graphql definition
-directive @join__owner(graph: join__Graph!)
-  on OBJECT
+directive @join__owner(graph: join__Graph!) on OBJECT
 ```
 
 Object types with keys MUST be owned by a subgraph. The owning subgraph:
```

Comment:
Does "object types with keys" mean "Any object type with at least one `@join__type` directive with non-null `key`"? If so we should just say that.  Is there a requirement that the given owner is one of those mentioned in a non-null `key` `@join__type`?  Also instead of "MUST be owned by a subgraph" I think we should say "must have a `@join__owner` directive". In general I think it's better if our MUST rules describe the structure of the schema document rather than abstract concept that we have to relate back to the document.

On the next, line, I don't know what "MUST be able to resolve all of the object's keys for any subgraph" means. Does it mean "For every `key` on every `@join__type` directive on the object, the `key` must be a valid selection set on the given type within the owning subgraph's schema" or something?

Similarly for the second bullet item. Though that also needs to explain that we're looking at every field to find these selection sets. Maybe the idea is that we first say "collect a bunch of selection sets from `key` and `requires`, then make sure that they are valid in the owner's schema"

Reply:
Fixed more or less